src/stores/simulations.ts

```diff
@@ -16,18 +26,53 @@ export const useSimulationStore = defineStore('simulation', () => {
     }
   }
 
+  function setCurrentSubStep(subStep: number) {
+    if (subStep < 0) {
+      currentSubStep.value = 0
+    } else if (subStep > 2) {
```

Comment:
Here of the next is step 3 and substep 3 , the substep is set to 2. It's not good

Reply:
You can just make this function on one line : currentSubstep.value = subStep
Same for function setCurrentStep